base/sparse/sparsematrix.jl

```diff
@@ -1264,7 +1264,7 @@ function find(p::Function, S::SparseMatrixCSC)
     end
     sz = size(S)
     I, J = _findn(p, S)
-    return sub2ind(sz, I, J)
+    return CartesianIndex.(I, J)
```

Comment:
With the new behavior of `find`, `findn` could probably be removed as it gives almost the same information (struct of arrays vs. array of structs).

This method could also be optimized a bit by not allocating `I` and `J` first (but it already does less work than before).